### PR TITLE
Added SpatialDropout1D

### PR DESCRIPTION
**Pull request by colincsl**
This is a straightforward modification of SpatialDropout2D (and similarly SpatialDropout3D) for 1D data.

### CODE REVIEW

**Comment by fchollet** [comment]
Tests are failing, please fix.

File: keras/layers/core.py [hunk 1]
```diff
@@ -96,6 +96,53 @@ def get_config(self):
...
+        p: float between 0 and 1. Fraction of the input units to drop.
+        dim_ordering: 'th' or 'tf'. In 'th' mode, the channels dimension
+            (the depth) is at index 1, in 'tf' mode is it at index 3.
+            It defaults to the `image_dim_ordering` value found in your
+            Keras config file at `~/.keras/keras.json`.
+            If you never set it, then it will be "tf".
+
+    # Input shape
+        3D tensor with shape:
+        `(samples, channels, timesteps)` if dim_ordering='th'
```
Comment (fchollet) [inline keras/layers/core.py]: There are no 1D layers that use this dim ordering. I would recommend removing support for it entirely for 'th' dim ordering here (see e.g. Conv1D).

File: keras/layers/core.py [hunk 2]
```diff
@@ -96,6 +96,49 @@ def get_config(self):
...
+    def __init__(self, p, dim_ordering='default', **kwargs):
+        if dim_ordering == 'default':
+            dim_ordering = K.image_dim_ordering()
+        assert dim_ordering in {'tf', 'th'}, 'dim_ordering must be in {tf, th}'
+        self.dim_ordering = dim_ordering
+        super(SpatialDropout1D, self).__init__(p, **kwargs)
+
+    def _get_noise_shape(self, x):
+        input_shape = K.shape(x)
+        if self.dim_ordering in ['th', 'tf']:
```
Comment (fchollet) [inline keras/layers/core.py]: Get rid of `dim_ordering`, as it isn't doing anything anymore.

**Comment by colincsl** [comment]
Hey, sorry, yeah I'll take care of this tomorrow. The CVPR deadline is tonight.

**Comment by RaffEdwardBAH** [comment]
No problem. Good luck on your paper!
